fggs/sum_product.py

```diff
@@ -99,10 +101,23 @@ def F(fgg: FGG, x: MultiTensor, inputs: MultiTensor, semiring: Semiring) -> Mult
     for n in x.shapes[0]:
         for rule in fgg.rules(n):
             tau_rule = sum_product_edges(fgg, rule.rhs.nodes(), rule.rhs.edges(), rule.rhs.ext, x, inputs, semiring=semiring)
-            Fx.add_single(n, tau_rule)
+            if tau_rule is not None: Fx.add_single(n, tau_rule)
     return Fx
 
 
+def print_duplicate(loc: str, xs: Sequence[Node], ys: Sequence[Node]) -> bool:
```

Comment:
Is this just for debugging? Does it need to stay?

Reply:
It's just for debugging. It doesn't need to stay. I just feel like it might be useful for future debugging...?